Src/FluentAssertions/Collections/GenericDictionaryAssertions.cs

```diff
@@ -242,6 +243,87 @@ public GenericDictionaryAssertions(IDictionary<TKey, TValue> dictionary) : base(
             return new AndConstraint<GenericDictionaryAssertions<TKey, TValue>>(this);
         }
 
+        /// <summary>
+        /// Assert that the current dictionary has the same number of elements as <paramref name="otherCollection" />.
+        /// </summary>
+        /// <param name="otherCollection">The other collection with the same expected number of elements</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<GenericDictionaryAssertions<TKey, TValue>> HaveSameCount(IEnumerable otherCollection, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(otherCollection, nameof(otherCollection), "Cannot compare dictionary count against a <null> collection.");
+
+            if (ReferenceEquals(Subject, null))
```

Comment:
Considering using `is null`.

Reply:
The reason `is null` isn't used in `CollectionAssertions.HaveSameCount` is because `TSubject` is not constrained to reference types and `is null` can only be used on reference types.

Reply:
SonarQube keeps telling me (in another project) to compare to `default(T)` ;-)